junit5/core/src/main/java/org/jboss/arquillian/junit5/ArquillianExtension.java

```diff
@@ -95,44 +92,44 @@ public void interceptTestMethod(Invocation<Void> invocation, ReflectiveInvocatio
                     .ifPresent(ExceptionUtils::throwAsUncheckedException);
         }
     }
-    
+
     @Override
     public void interceptBeforeEachMethod(Invocation<Void> invocation,
       ReflectiveInvocationContext<Method> invocationContext, ExtensionContext extensionContext) throws Throwable {
-      if (isInsideArquillian.test(extensionContext)) {
+      if (isInsideArquillian.test(extensionContext) || isRunAsClient(extensionContext)) {
         invocation.proceed();
       } else {
         invocation.skip();
       }
     }
-    
+
     @Override
     public void interceptAfterEachMethod(Invocation<Void> invocation,
       ReflectiveInvocationContext<Method> invocationContext, ExtensionContext extensionContext) throws Throwable {
-      if (isInsideArquillian.test(extensionContext)) {
+      if (isInsideArquillian.test(extensionContext) || isRunAsClient(extensionContext)) {
         invocation.proceed();
       } else {
         invocation.skip();
       }
     }
-    
+
     @Override
     public void interceptBeforeAllMethod(Invocation<Void> invocation,
       ReflectiveInvocationContext<Method> invocationContext, ExtensionContext extensionContext) throws Throwable {
       if (isInsideArquillian.test(extensionContext)) {
-        invocation.proceed();
-      } else {
         invocation.skip();
+      } else {
+        invocation.proceed();
```

Comment:
Why are we reversing the logic here? We don't want to have `@BeforeAll` executed as Arquillian test?

Reply:
Seems to me we are going backwards  to the same issue we had two years back - it depends. It makes sense for both use cases, sometimes.

I think we need another annotation for this and decide what should be the default - run on the client side or on the server side? And warn before undefined order if both variants combined (multiple BeforeEach annotated methods). 

Or am I wrong?

Reply:
Not the same case David :). This is the simplest case

Reply:
@bartoszmajsak the logic is reversed to accomplish the fix for the issue. These methods have to be only executed on the client and not in container as they are executed that way in JUnit 4

I’ll try to add the test for this as well

Reply:
I have added the tests. However, they are excluded for now. This requires integration test setup with app server as far as I am aware, and it's not done in Arquillian just yet, so I am not sure how to introduce that kind of tests.
Saying that, it's there and works when enabled :)